### ISSUE TITLE
Community to review pin requests

### ISSUE THREAD

**Issue by ghost** [VictorBjelkholm/openipfs#13]
The project is a great idea! Isn't there though the problem of a malicious user pinning trash files to the network. Shouldn't there exist a webapp on top of openipfs which would be run by the community (like stackoverflow) where users would submit files for pinning and according to some guidelines a community would review that request and either allow or deny it?

**Comment by victorb** [comment]
Oh, this is a cool idea! Basically like a queue of pins, ability to review
them and once it passes a certain amount of "upvotes" or "oks" by people,
then it'll be pinned.

I like the idea and is worth discussing for sure.

Thanks for the feedback, keep it coming :)

Sincerely,
Victor Bjelkholm
(+34) 672 15 90 89

**Comment by ghost** [comment]
Yeah, like a user curated global digital library. What wikipedia does for an encyclopedia we could do for a library. On that library we could keep enough copies of anything that's worth keeping around for future generations.

At a first glance I guess we would need:
1)Ipfs for storage
2)A system to check how many copies of each item exist on the network and creating enough of them when needed
3)A distributed webapp for users to participate in the curation of the library but also to be able to see each item 
4)A distributed search engine for searching the items

**Comment by pokeball99** [comment]
@anastasiospapanikolaou ,not just a library,for a internet :)
